// @kazion/fcm-node-http/lib/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AndroidMessagePriority;
(function (AndroidMessagePriority) {
    AndroidMessagePriority["NORMAL"] = "NORMAL";
    AndroidMessagePriority["HIGH"] = "HIGH";
})(AndroidMessagePriority || (AndroidMessagePriority = {}));
var NotificationPriority;
(function (NotificationPriority) {
    NotificationPriority["PRIORITY_UNSPECIFIED"] = "PRIORITY_UNSPECIFIED";
    NotificationPriority["PRIORITY_MIN"] = "PRIORITY_MIN";
    NotificationPriority["PRIORITY_LOW"] = "PRIORITY_LOW";
    NotificationPriority["PRIORITY_DEFAULT"] = "PRIORITY_DEFAULT";
    NotificationPriority["PRIORITY_HIGH"] = "PRIORITY_HIGH";
    NotificationPriority["PRIORITY_MAX"] = "PRIORITY_MAX";
})(NotificationPriority || (NotificationPriority = {}));
var Visibility;
(function (Visibility) {
    Visibility["VISIBILITY_UNSPECIFIED"] = "VISIBILITY_UNSPECIFIED";
    Visibility["PRIVATE"] = "PRIVATE";
    Visibility["PUBLIC"] = "PUBLIC";
    Visibility["SECRET"] = "SECRET";
})(Visibility || (Visibility = {}));
